### ISSUE TITLE
Issue -inf Likelihood with any bmodels/rmodels

### ISSUE THREAD

**Issue by MaximePolicarpo** [fgvieira/badirate#5]
Hi,

I am currently trying to run BadiRate but I always end up with the same error : 

The command line always return the message : "WARN: Try using a more complex model, or changing the starting values. See the -rmodel, -bmodel and -start_val options" .

I tried to modify the -rmodel, the -bmodel, the -ep option and the -start_val value but it print this WARN every time and my output file always look like this : 

OUTPUT

	##Family Turnover Rates
		#Likelihood: -inf

	##Execution time (seconds): 562
END OUTPUT

I attached two files : My species tree and the table with the gene count per OG for each species.
And here is an example of a command line I tried : 

perl BadiRate.pl -anc -treefile Species_Tree_rooted_ultrametric_newick.txt  -sizefile table_OGs_protein_counts.txt –bmodel FR –ep CMAP -rmodel BDI –family -unobs -start_val 1 –root_dist 0 > output.bd 

Would anyone know why I never get any results ? 

Thanks for any help provided,

Maxime Policarpo
[table_OGs_protein_counts.txt](https://github.com/fgvieira/badirate/files/5505263/table_OGs_protein_counts.txt)
[Species_Tree_rooted_ultrametric_newick.txt](https://github.com/fgvieira/badirate/files/5505264/Species_Tree_rooted_ultrametric_newick.txt)

**Comment by fgvieira** [comment]
Hi Maxime,

from what you are telling me, it seems that your data is too heterogeneous for the model you are using. That is, you have gene families with very different rates and no model can explain them. Usually, the way to go is to use more complex models but, sometimes, no model can fit the data. This can happen, for example, if you have transposable elements in your dataset, as these have tipically much higher rates than other genes.

I'd remove all transposable element families from the dataset before running BadiRate. If you are very interested in them, then I'd try spliting the dataset in two and analyze them separately.

**Comment by MaximePolicarpo** [comment]
Hi Filipe,

My dataset is only composed by olfactory receptor, than indeed have very different rates of duplications/losses across my species. Would you suggest to split the species tree dataset or split the count table (For example if I have 30 Ortholog groups, perform an analysis with the first 15 OG and a second analysis with the 15 others ?)

Thanks a lot for your help,

Maxime

**Comment by fgvieira** [comment]
Have you tried a FR model?

**Comment by MaximePolicarpo** [comment]
Yes, I almost tried every combinations of -bmodel and -rmodel possible :(

**Comment by MaximePolicarpo** [comment]
In fact, even when I try with only 1 sub-family of my file (OG_1), I always end up with the same warning. I tried to launch BadiRate with the same species tree but with another dataset, and this time It worked fine. I don't really know how to handle this problem ..

**Comment by fgvieira** [comment]
Did you try with only one family (e.g. OG_1) but just using parsimony?

**Comment by MaximePolicarpo** [comment]
Yes, I tried : 

/home/casane/perl5/perlbrew/perls/perl-5.16.0/bin/perl /home/casane/badirate-1.35/BadiRate.pl -anc -treefile Species_Tree_rooted_ultrametric_newick.txt -sizefile table_OGs_protein_counts_OG1.txt –bmodel FR –ep CWP -rmodel BDI -start_val 1 –root_dist 0 > output.bd

/home/casane/perl5/perlbrew/perls/perl-5.16.0/bin/perl /home/casane/badirate-1.35/BadiRate.pl -anc -treefile Species_Tree_rooted_ultrametric_newick.txt -sizefile table_OGs_protein_counts_OG1.txt –bmodel FR –ep CWP -rmodel BDI -start_val 0 –root_dist 0 > output.bd

Still the same warning message :/

**Comment by YiyanYang0728** [comment]
I also encountered this issue. And the gene family in my dataset is not transposable elements-related. I tried all the combinations with ep=ML and ended up with this same warning (even FR did not work). 
At last, I modified the original code (Badirate.pl) on line 1894-1895 in the function for BDI model, like this:
```
1894         #$error=2;
1895         return -$y; # "-inf";
```
Then I got some results but with Likelihood=-1e+26.

I want to ask if this result is usable or not. Is it possible to give a prior file which is able to fit highly heterogeneous data? I'd really appreciate your help and suggestion.

Thanks,
Yiyan

**Comment by fgvieira** [comment]
@YiyanYang0728 not sure if you have been contacted already, but I'd suggest contacting the main author directly (Pablo Librado), as stated in the README.
